apps/api/src/app/events/e2e/process-subscriber.e2e.ts

```diff
@@ -111,4 +93,81 @@ describe('Trigger event - process subscriber /v1/events/trigger (POST)', functio
     expect(createdSubscriber.lastName).to.equal(payload.lastName);
     expect(createdSubscriber.email).to.equal(payload.email);
   });
+
+  it('should send only email trigger second time based on the subscriber preference', async function () {
+    const payload: ISubscribersDefine = {
+      subscriberId: session.subscriberId,
+      firstName: 'New Test Name',
+      lastName: 'New Last of name',
+      email: 'newtest@email.novu',
+    };
+
+    await triggerEvent(session, template, payload);
+
+    await session.awaitRunningJobs(template._id);
+
+    const widgetSubscriber = await subscriberRepository.findBySubscriberId(
+      session.environment._id,
+      session.subscriberId
+    );
+
+    let message = await messageRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+      _subscriberId: widgetSubscriber._id,
+    });
+
+    expect(message.length).to.equal(2);
+
+    const updateData = {
+      channel: {
+        type: ChannelTypeEnum.IN_APP,
+        enabled: false,
+      },
+    };
+
+    await updateSubscriberPreference(updateData, session.subscriberToken, template._id);
+
+    await triggerEvent(session, template, payload);
+
+    await session.awaitRunningJobs(template._id);
+
+    message = await messageRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+      _subscriberId: widgetSubscriber._id,
+    });
+
+    expect(message.length).to.equal(3);
```

Comment:
I have learned before that it is nice to test so the messages you expect is in the message array...

Reply:
Basically here I wanted to show that the first trigger sent 2 messages and the second (after disabling the "in-app") only 1 triggered (so the sum is 3)

Do you mean I should test the message's context as well?

Reply:
Nah type may be fine